### ISSUE TITLE
EnforceCodeStyleInBuild reports suggestions as errors in 8.0.300

### ISSUE THREAD

**Issue by glen-nicol** [dotnet/roslyn#73974]
### Describe the bug
I recently updated Visual studio and it updated my .net SDK from 8.0.205 to 8.0.300 and since doing so my build fails due to IDE style errors because I have EnforceCodeStyleInBuild enabled. The IDEXXX and csharp_style_* rules that are reported are all configured as suggestions however and should not be reported as errors. I can workaround the bug by using global.json to specify 8.0.205 and disable rollforward.

NOTE: I also tried 8.0.302 with the same result.

### To Reproduce
I tried to reproduce it with a simple project with no success. This is a fairly complicated project that has evolved over time with .net so I wouldn't be surprised if I have a few settings that are stale. I could probably share my .editorconfig file as reference.

an example of one of the rules that is falsely reported as error:
```csharp_style_deconstructed_variable_declaration = true:suggestion```

### Further technical details
```
dotnet --info
.NET SDK:
 Version:           8.0.300
 Commit:            326f6e68b2
 Workload version:  8.0.300-manifests.ca8b4b2d
 MSBuild version:   17.10.4+10fbfbf2e

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19045
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.300\

.NET workloads installed:
 [aspire]
   Installation Source: VS 17.10.35004.147
   Manifest Version:    8.0.0/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.aspire\8.0.0\WorkloadManifest.json
   Install Type:        FileBased


Host:
  Version:      8.0.6
  Architecture: x64
  Commit:       3b8b000a0e

.NET SDKs installed:
  3.1.426 [C:\Program Files\dotnet\sdk]
  6.0.321 [C:\Program Files\dotnet\sdk]
  8.0.205 [C:\Program Files\dotnet\sdk]
  8.0.300 [C:\Program Files\dotnet\sdk]
  8.0.302 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.18 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.30 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.18 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  C:\git\<redacted>\global.json

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

Visual Studio 22 17.10.2

**Comment by arkalyanms** [comment]
Parking on Sam until we tweak the assembly load to match.

**Comment by sharwell** [comment]
Can you use Visual Studio's Report a Problem tool to file a report on this with additional information? The specific information we need is a copy of the **msbuild.binlog** file produced when you build your project from the command line with the `-bl` flag.

**Comment by glen-nicol** [comment]
> Can you use Visual Studio's Report a Problem tool to file a report on this with additional information? The specific information we need is a copy of the **msbuild.binlog** file produced when you build your project from the command line with the `-bl` flag.

I can probably provide that if it will give me the chance to redact portions of it. Are there specific pieces of that file you're looking for? I imagine there will be plenty of type names included that I need to redact.

**Comment by sharwell** [comment]
I'd be looking at the inputs to the `$task Csc` to determine the complete set of AnalyzerConfigFiles passed to the compiler, along with any command line arguments related to the diagnostic you're seeing here. One of them is setting the severity to warning or error. It shouldn't be difficult to locate the source of the severity setting using the [MSBuild Log Viewer](https://msbuildlog.com/).

**Comment by glen-nicol** [comment]
So using MSBuild Log Viewer I was able to find a project that had one error. Figured that would be a simple case to work through. It also happens to be the first error in the build for what that's worth.

```...\HttpLib\BasicHttpProvider.cs(9,66): error IDE0090: 'new' expression can be simplified```

And in the command line args for compiling that dll I found only these three analyzerconfig arguments:

```
/analyzerconfig:C:\git\sln\.editorconfig
/analyzerconfig:obj\Debug\netstandard2.0\HttpLib.GeneratedMSBuildEditorConfig.editorconfig
/analyzerconfig:C:\Users\nicolg\.nuget\packages\microsoft.codeanalysis.netanalyzers\8.0.0\buildTransitive\config\analysislevel_8_recommended.globalconfig
```

None which mention IDE0090. 

Have I misunderstood how to interpret the log? Are there other places where severity can be set? I admit I don't know all the arguments for csc, but I didn't see anything obvious. Certainly no IDE0090 explicitely.

**Comment by sharwell** [comment]
Do any of those files list `csharp_style_implicit_object_creation_when_type_is_apparent`?

**Comment by glen-nicol** [comment]
yes, my main .editorconfig has this:

```csharp_style_implicit_object_creation_when_type_is_apparent = true:suggestion```

hmm, did the legacy ```true:suggestion``` severity formatting finally get pulled and now causing a fuss?

**Comment by glen-nicol** [comment]
I guess the old formatting isn't a problem but not having the new format is. Adding the IDE0090 rule correctly sets the severity and gets rid of the error. Is there a tool that can go through all those and duplicate them? I have quite a few set using the old format.

**Comment by glen-nicol** [comment]
I ended up just reviewing all the errors in VS and using its UI to configure them to suggestions. Thankfully those worked reliably this time! Thanks @sharwell for your help and thanks for all your hard work on dotnet!

**Comment by sharwell** [comment]
> hmm, did the legacy true:suggestion severity formatting finally get pulled and now causing a fuss?

The `true:suggestion` syntax will start to take effect for .NET 9 SDK with `AnalysisLevel` set to 9 or greater. It may also work that way in 8.0.300 if you manually set `AnalysisLevel` to 9 or greater.

> ```
> csharp_style_implicit_object_creation_when_type_is_apparent = true:suggestion
> ```

This would not be sufficient to trigger an error. The default severity for IDE0090 is suggestion, and TreatWarningsAsErrors only applies to warnings and errors. If an error is reported in the build, _something_ has reconfigured IDE0090 from its default severity of suggestion to either error, or to warning with TreatWarningsAsErrors enabled.

**Comment by glen-nicol** [comment]
I have this setup:

```
<EnableNETAnalyzers>true</EnableNETAnalyzers>
<AnalysisMode>All</AnalysisMode>
<AnalysisLevel>latest-recommended</AnalysisLevel>
```

And I even commented all those out and I still got errors instead of suggestions (This was before I set the rules).

Would 8.0.300 target 9 with latest-recommended? I wouldn't have expected that myself.

**Comment by sharwell** [comment]
> And I even commented all those out and I still got errors instead of suggestions (This was before I set the rules).

I'm not sure I'll be able to resolve this without a concrete repro (either a sample solution or build log submitted through the feedback system and not posted publicly).

> Would 8.0.300 target 9 with latest-recommended?

No, 8.0.300 defines `_LatestAnalysisLevel` as 8.0.
